storage: move many commands to `batcheval`

tbg opened the pull request:
This spares `endTransaction` which merits its own PR due to the commit
triggers.

It also does not yet move the span declaration methods, but nothing is
blocking that.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/19900)
<!-- Reviewable:end -->

bdarnell commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>, but if we're going to have a file per RPC I think the filenames should always match the RPC name (so I'd rather have `cmd_*_txn.go` instead of `cmd_txn_*.go`). Alternately, we could group "related" rpcs together into a handful of files (one file for the lease commands, one for txn stuff, one for basic KV ops...)

---

Reviewed 35 of 35 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion, all commit checks successful.

---

*[pkg/storage/batcheval/eval_txn_push.go, line 1 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/19900#-KyRc0Dfsd4qQ_g_aovw:-KyRc0Dfsd4qQ_g_aovx:bcy40l) ([raw file](https://github.com/cockroachdb/cockroach/blob/b97d41114aab963daa8d0cc3e96803c8c3db9e75/pkg/storage/batcheval/eval_txn_push.go#L1)):*
> ```Go
> // Copyright 2014 The Cockroach Authors.
> ```

This file doesn't fit the pattern (`eval_` vs `cmd_`). 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/19900#-:-KyRb_4SsfQe2Znt_DJw:besiwmm)*
<!-- Sent from Reviewable.io -->

nvanbenschoten commented:
Reviewed 15 of 35 files at r1.
Review status: all files reviewed at latest revision, 2 unresolved discussions, all commit checks successful.

---

*[pkg/storage/replica_command.go, line 63 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/19900#-KyRhda-GmORH2jOr2Zt:-KyRhda-GmORH2jOr2Zu:b-o22sck) ([raw file](https://github.com/cockroachdb/cockroach/blob/b97d41114aab963daa8d0cc3e96803c8c3db9e75/pkg/storage/replica_command.go#L63)):*
> ```Go
> )
> 
> // A Command is the implementation of a single request within a BatchRequest.
> ```

Do you think the `Command` type and the `commands` map should be moved to the new `batcheval` package? That way, a new command doesn't even need to worry about this file. Conceptually, this means that a new command doesn't need to worry about how it will be evaluated by a `Replica`.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/19900)*
<!-- Sent from Reviewable.io -->

tbg committed:
storage: move many commands to `batcheval`

This spares `endTransaction` which merits its own PR due to the commit
triggers.

It also does not yet move the span declaration methods, but nothing is
blocking that.

tschottdorf commented:
TFTR!
I went for filename==RPC name for now.

---

Review status: 28 of 35 files reviewed at latest revision, 2 unresolved discussions.

---

*[pkg/storage/replica_command.go, line 63 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/19900#-KyRhda-GmORH2jOr2Zt:-KyRxguQFnaRvICWpY76:b-58jni1) ([raw file](https://github.com/cockroachdb/cockroach/blob/b97d41114aab963daa8d0cc3e96803c8c3db9e75/pkg/storage/replica_command.go#L63)):*
<details><summary><i>Previously, nvanbenschoten (Nathan VanBenschoten) wrote…</i></summary><blockquote>

Do you think the `Command` type and the `commands` map should be moved to the new `batcheval` package? That way, a new command doesn't even need to worry about this file. Conceptually, this means that a new command doesn't need to worry about how it will be evaluated by a `Replica`.
</blockquote></details>

Yes, I definitely want to move these over, just haven't done so in this PR.

---

*[pkg/storage/batcheval/eval_txn_push.go, line 1 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/19900#-KyRc0Dfsd4qQ_g_aovw:-KyRyGjHn1lUereA9Uo9:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/b97d41114aab963daa8d0cc3e96803c8c3db9e75/pkg/storage/batcheval/eval_txn_push.go#L1)):*
<details><summary><i>Previously, bdarnell (Ben Darnell) wrote…</i></summary><blockquote>

This file doesn't fit the pattern (`eval_` vs `cmd_`). 
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/19900#-:-KyRyUC08rf7kGyHu5qN:b-fx7sr0)*
<!-- Sent from Reviewable.io -->
